### PR TITLE
New SPO Features Updates for 4.13

### PR DESCRIPTION
**Pull request by GroceryBoyJr**
This PR was superceded by https://github.com/openshift/openshift-docs/pull/59960 due to new information found and published by Andrew Taylor. @GroceryBoyJr is closing it as it is no longer needed.

Version(s): 4.13+

Issue: New features for SPO 

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by GroceryBoyJr** [comment]
@sheriff-rh Looking for your initial review and your suggestion of where to slot in these two new snippets.

**Comment by GroceryBoyJr** [comment]
@jhrozek here's my first pass at two snippets describing the new features for SPO. Looking for your SME review please.

**Review by jhrozek — CHANGES_REQUESTED** [review]
Thank you, mostly looking good! I left some comments inline

**Comment by sheriff-rh** [comment]
@GroceryBoyJr I think the memory optimization snippet could go somewhere (anywhere, really) in the [advanced tasks](https://docs.openshift.com/container-platform/4.12/security/security_profiles_operator/spo-advanced.html) assembly. The auto select of selinuxd image should go in the [spo-selinux](https://docs.openshift.com/container-platform/4.12/security/security_profiles_operator/spo-selinux.html) assembly, probably at the top.

File: modules/spo-memory-optimize.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+== Configure the SPO daemon resource requirements
+
+The default resource requirements of the daemon container can be adjusted by using the field daemonResourceRequirements from the SPOD configuration as follows:
+
++
+[source,terminal]
+----
+$ oc -n security-profiles-operator patch spod \
+    spod --type=merge -p '{"spec":{"allowedSyscalls": ["exit", "exit_group", "futex", "nanosleep"]}}'
+----
```
[the remark this answers is not in the document]
Comment (GroceryBoyJr) [reply modules/spo-memory-optimize.adoc]: @jhrozek Done, thank you!

File: modules/spo-memory-optimize.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Also, when the list of allowed syscalls is modified in the spod configuration, the operator will automatically identify the already installed profiles which are not compliant and remove them.
+
+== Set a custom priority class name for SPO daemon
+
+The default priority class name of the spod daemon pod is set to system-node-critical. A custom priority class name can be configured in the SPOD configuration by setting a value in the priorityClassName filed.
+
++
+[source,terminal]
+----
+$ oc  -n security-profiles-operator patch spod \
```
[the remark this answers is not in the document]
Comment (GroceryBoyJr) [reply modules/spo-memory-optimize.adoc]: @jhrozek done, thank you!

File: snippets/spo-memory-optimize.adoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+The default resource requirements of the daemon container can be adjusted by using the field daemonResourceRequirements from the SPOD configuration as follows:
+
++
+[source,terminal]
+----
+$ oc -n openshift-security-profiles patch spod spod --type merge -p \
+'{"spec":{"daemonResourceRequirements": {"requests": {"memory": "256Mi", "cpu": "250m"}, "limits": {"memory": "
+----
+
+After this command the operator will only install the seccomp profiles which have a subset of syscalls defined into the allowed list. All profiles not complying with this rule will be rejected.
```
Comment (jhrozek) [inline snippets/spo-memory-optimize.adoc:58]: This and the following line are probably copy-pasted from somewhere else right?

**Review by jhrozek — COMMENTED** [review]
One more nit otherwise I'm happy

File: snippets/spo-memory-optimize.adoc (hunk 3 above)
Comment (GroceryBoyJr) [reply snippets/spo-memory-optimize.adoc:58]: @jhrozek They came from the upstream reference you cited. I'm glad to delete them if you wish!

Comment (GroceryBoyJr) [reply snippets/spo-memory-optimize.adoc:58]: @jhrozek thank you!

**Comment by GroceryBoyJr** [comment]
This PR was superceded by https://github.com/openshift/openshift-docs/pull/59960 due to new information found and published by Andrew Taylor. @GroceryBoyJr is closing it as it is no longer needed.

**Pull request closed without merging**